apps/test/unit/accounts/ChangeUserTypeModalTest.jsx

```diff
@@ -1,215 +1,175 @@
-import {mount} from 'enzyme'; // eslint-disable-line no-restricted-imports
+import {render, screen, fireEvent} from '@testing-library/react';
 import React from 'react';
-import sinon from 'sinon'; // eslint-disable-line no-restricted-imports
 
 import ChangeUserTypeModal from '@cdo/apps/accounts/ChangeUserTypeModal';
-import Button from '@cdo/apps/legacySharedComponents/Button';
-import i18n from '@cdo/locale';
 
-import {expect} from '../../util/deprecatedChai'; // eslint-disable-line no-restricted-imports
+// Mock i18n strings used in the components
+jest.mock('@cdo/locale', () => {
+  const customStrings = {
+    changeUserTypeModal_save_teacher: () => 'Save Teacher',
+    cancel: () => 'Cancel',
+    saving: () => 'Saving...',
+    changeUserTypeModal_title: () => 'Change User Type',
+    changeUserTypeModal_unexpectedError: () => 'An unexpected error occurred.',
+    changeUserTypeModal_email_isRequired: () => 'Email is required.',
+    changeUserTypeModal_email_invalid: () => 'Invalid email.',
+    changeUserTypeModal_emailOptIn_description: () =>
+      'Email opt-in description.',
+    changeUserTypeModal_emailOptIn_isRequired: () =>
+      'Email opt-in is required.',
+  };
+  return {
+    __esModule: true,
+    default: new Proxy(customStrings, {
+      get: (target, prop) => {
+        return prop in target ? target[prop] : () => prop;
+      },
+    }),
+  };
+});
 
 describe('ChangeUserTypeModal', () => {
-  let wrapper;
-
-  const EMAIL_SELECTOR = 'input[type="email"]';
-  const EMAIL_OPT_IN_SELECTOR = 'select';
-
   const DEFAULT_PROPS = {
-    handleSubmit: () => {},
-    handleCancel: () => {},
+    handleSubmit: jest.fn(() => Promise.resolve()),
+    handleCancel: jest.fn(),
   };
 
-  // Helpers for selecting particular elements/components
-  const emailInput = wrapper => wrapper.find(EMAIL_SELECTOR);
-  const emailOptInSelect = wrapper => wrapper.find(EMAIL_OPT_IN_SELECTOR);
-  const submitButton = wrapper =>
-    wrapper
-      .find(Button)
-      .filterWhere(
-        n => n.prop('text') === i18n.changeUserTypeModal_save_teacher()
-      );
-  const cancelButton = wrapper =>
-    wrapper.find(Button).filterWhere(n => n.prop('text') === i18n.cancel());
+  const renderComponent = (props = {}) => {
+    return render(<ChangeUserTypeModal {...DEFAULT_PROPS} {...props} />);
+  };
 
-  beforeEach(() => {
-    wrapper = mount(<ChangeUserTypeModal {...DEFAULT_PROPS} />);
-  });
+  const getEmailInput = () => screen.getByRole('textbox', {name: /email/i});
+  const getEmailOptInSelect = () =>
+    screen.getByRole('combobox', {name: /email opt-in/i});
+  const getSubmitButton = () =>
+    screen.getByRole('button', {name: /save teacher/i});
+  const getCancelButton = () => screen.getByRole('button', {name: /cancel/i});
 
-  afterEach(() => {
-    wrapper.unmount();
+  it('renders the modal with the correct title', () => {
+    renderComponent();
+    expect(screen.getByText('Change User Type')).toBeInTheDocument();
```

Comment:
```suggestion
    screen.getByText('Change User Type')
```
screen.getByText will automatically check if it is in the document and error out if it is not found. We don't need the expect and we actually get a better error message if we remove it. 

Similar for all `toBeInTheDocument` below.

Reply:
Having a better error message sounds good.
On the other hand, I like having a consistent test structure where it's clear what is expected with at least an `expect` statement per test case.

Reply:
As per our discussion and [this article](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions). 
I'm going to leave the the test as I initially proposed with an explicit `expect` statement.